tests/test_accelerator.py

```diff
@@ -49,3 +61,83 @@ def test_free_memory_dereferences_prepared_components(self):
         self.assertTrue(len(accelerator._schedulers) == 0)
         self.assertTrue(len(accelerator._dataloaders) == 0)
         AcceleratorState._reset_state()
+
+    def test_save_load_model(self):
```

Comment:
This test is a bit unrelated to the PR, but think this is a useful test for the accelerator class in general.